File: Q_657.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        
        int i,ver=0,hor=0;
        
        for(i=0;i<moves.size();i++){
            switch(moves[i]){
                case 'U':{
                    ver++;
                    break;
                }
                case 'D':{
                    ver--;
                    break;
                }
                case 'L':{
                    hor--;
                    break;
                }
                case 'R':{
                    hor++;
                    break;
                }
            }
        }
        if(ver==0 && hor==0)
            return true;
        else
            return false;
        
    }
};

File: python/Q_657.py
class Solution:
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        count1 = 0;
        count2 = 0;
        mymap = {'U': 1, 'D': -1, 'L': 1, 'R': -1}
        for c in moves:
            if c in ['L', 'R']:
                count2 = count2 + mymap[c]
            elif c in ['U', 'D']:
                count1 = count1 + mymap[c]
        if count1 == 0 and count2 == 0:
            return True
        return False
        
            
         
